src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -168,21 +171,103 @@ trait EntityComponent {
       allEntitiesAction.flatMap(cloneEntities(destWorkspaceContext, _))
     }
 
+    def insertEntitiesSql(destWorkspaceContext: SlickWorkspaceContext, entities: Map[Entity, UUID]) = {
+      val workspaceId = destWorkspaceContext.workspaceId
+
+      val insertBatches = entities.map { case (entity, entityId) =>
+        sql"($entityId, ${entity.name}, ${entity.entityType}, $workspaceId)"
+      }.grouped(1000).toSeq
+
+      DBIO.seq(insertBatches.map { insertBatch =>
+        val prefix = sql"insert into ENTITY (id, name, entity_type, workspace_id) values "
+        val suffix = insertBatch.reduce { (a, b) =>
+          concatSqlActionsWithDelim(a, b, ",")
+        }
+        concatSqlActions(prefix, suffix).as[Int]
+      }:_*)
+    }
+
+    def insertAttributesSql(destWorkspaceContext: SlickWorkspaceContext, attrsWithEntityIdMap: Seq[(UUID, (String, Attribute), UUID, Option[Int])]) = {
```

Comment:
I think this case logic is in marshalAttribute, call that to create an AttributeRecord then you don't need the case here.